core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -158,9 +157,12 @@ public void sendMessage(RespT message) {
       InputStream resp = method.streamResponse(message);
       stream.writeMessage(resp);
       stream.flush();
+    } catch (RuntimeException e) {
+      close(Status.fromThrowable(e), new Metadata());
+      throw e;
     } catch (Throwable t) {
       close(Status.fromThrowable(t), new Metadata());
-      throw Throwables.propagate(t);
+      throw new RuntimeException(t);
```

Comment:
Wrapping an Error in a RuntimeException can cause the world to end but you not know why, because the receiver of the RuntimeException "handled" it or passed it to some future than someone ignored. Errors need to stay as Errors.